test/integration/tlspolicy_controller_test.go

```diff
@@ -170,6 +170,32 @@ var _ = Describe("TLSPolicy", Ordered, func() {
 				}, time.Second*10, time.Second).Should(BeEmpty())
 			})
 
+			It("should create certificate when TLS is present", func() {
+				certNS := gatewayv1beta1.Namespace(testNamespace)
+				patch := client.MergeFrom(gateway.DeepCopy())
+				gateway.Spec.Listeners[0].TLS = &gatewayv1beta1.GatewayTLSConfig{
+					Mode: Pointer(gatewayv1beta1.TLSModeTerminate),
+					CertificateRefs: []gatewayv1beta1.SecretObjectReference{
+						{
+							Name:      "test-tls-secret",
+							Namespace: &certNS,
+						},
+					},
+				}
+				Expect(k8sClient.Patch(ctx, gateway, patch)).To(BeNil())
+				Eventually(func() bool {
+					certList := &certmanv1.CertificateList{}
+					err := k8sClient.List(ctx, certList, &client.ListOptions{Namespace: testNamespace})
+					Expect(err).ToNot(HaveOccurred())
+					for _, cert := range certList.Items {
+						if cert.Name == "test-tls-secret" {
+							return true
+						}
+					}
+					return false
+				}, time.Second*10, time.Second).Should(BeTrue())
```

Comment:
Most of these Eventually/Consistently steps now return an `error` instead of a `bool`. This is to make it easier to see what the error is when/if it fails.

Also, since this is just expecting a cert to exist with a particular name, could it not be re-written as:

```
Eventually(func() error {
	cert := &certmanv1.Certificate{}
	return k8sClient.Get(ctx, client.ObjectKey{Name: "test-tls-secret", Namespace: testNamespace}, cert)
}, time.Second*10, time.Second).Should(BeNil())
```

Reply:
yeah, good point, I did it the other way as I wasn't certain what the name was going to be, but now it is known your suggestion is much tidier. I'll update it